src/librustc_mir/build/scope.rs

```diff
@@ -94,17 +94,31 @@ use rustc::ty::subst::{Kind, Subst};
 use rustc::ty::{Ty, TyCtxt};
 use rustc::mir::*;
 use rustc::mir::transform::MirSource;
-use syntax_pos::Span;
+use syntax_pos::{self, Span};
 use rustc_data_structures::indexed_vec::Idx;
 use rustc_data_structures::fx::FxHashMap;
 
+#[derive(Debug)]
+pub struct DivergePathState {
+    pub(crate) block: BasicBlock,
```

Comment:
considering how much `pub` proliferates in much of the compiler, I was figuring a `pub(crate)` was a conservative choice here...  anyway I'll look into priv.